kodilla-hibernate/src/test/java/com/kodilla/hibernate/manytomany/dao/CompanyDaoTestSuite.java

```diff
@@ -126,5 +126,78 @@ public void testNamedQueries() {
         companyDao.deleteById(greyMatterId);
         companyDao.deleteById(dataMaestersId);
     }
+
+    @Test
+    public void testNamedQueryRetrieveEmployeeWithGivenPhraseInLastName() {
+        //Given
+        Employee johnSmith = new Employee("John", "Smarth");
+        Employee stephanieClarckson = new Employee("Stephanie", "Clarckson");
+        Employee lindaKovalsky = new Employee("Linda", "Kovalsky");
+        Employee marthaSmith = new Employee("Martha", "Smith");
+
+        employeeDao.save(johnSmith);
+        int johnSmithId = johnSmith.getId();
+        employeeDao.save(stephanieClarckson);
+        int stephanieClarcksonId = stephanieClarckson.getId();
+        employeeDao.save(lindaKovalsky);
+        int lindaKovalskyId = lindaKovalsky.getId();
+        employeeDao.save(marthaSmith);
+        int marthaSmithId = marthaSmith.getId();
+
+        //When
+        List<Employee> employeesWithGivenPhraseInLastName = employeeDao.retrieveEmployeeWithGivenPhraseInLastName("%sm%");
+        List<String> fullNames = employeesWithGivenPhraseInLastName.stream()
+                .map(Employee::getLastName)
+                .collect(Collectors.toList());
+
+        //Then
+        Assert.assertEquals(2, employeesWithGivenPhraseInLastName.size());
+        Assert.assertTrue(fullNames.contains("Smith"));
+        Assert.assertTrue(fullNames.contains("Smarth"));
+        Assert.assertFalse(fullNames.contains("Clarckson"));
+
+        //CleanUp
+        employeeDao.deleteById(johnSmithId);
+        employeeDao.deleteById(stephanieClarcksonId);
+        employeeDao.deleteById(marthaSmithId);
+        employeeDao.deleteById(lindaKovalskyId);
+    }
+
+    @Test
+    public void testNamedQueryRetrieveCompaniesWithGivenPhraseInName() {
```

Comment:
Zastanawiam się, czy ten test nie jest nadmiarowy.  Właściwie to samo testujesz w fasadzie (oczywiście z różnicą co do %). Po to wystawiasz fasadę, aby to z niej korzystali użytkownicy. Jeśli całe zachowanie systemu jesteś w stanie przetestować na fasadzie, to testowanie tych głębszych warstw właściwie nie przynosi dodatkowej wartości, a będziesz miała więcej przeróbek w przypadku np. wymiany technologii bazodanowej.